src/main/java/spoon/support/sniper/internal/AbstractSourceFragmentContextCollection.java

```diff
@@ -68,6 +73,17 @@ public boolean knowsHowToPrint(PrinterEvent event) {
 		throw new SpoonException("Unexpected PrintEvent: " + event.getClass());
 	}
 
+	private boolean checkIfPartOfRole(CtRole role) {
+		List<SourceFragment> elementSourceFragments = childFragments.stream()
+				.filter(fragment -> fragment instanceof ElementSourceFragment)
+				.collect(Collectors.toList());
+		for (SourceFragment sourceFragment: elementSourceFragments) {
+			ElementSourceFragment elementSourceFragment = ((ElementSourceFragment) sourceFragment);
+			return elementSourceFragment.getRoleInParent() == role;
+		}
+		return false;
+	}
```

Comment:
This is really hard to read and should be refactored in a single loop. A loop that always returns and doesn't loop is kinda unexpected.